pkg/scaling/scale_handler_raw_metrics.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2025 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scaling
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	"github.com/kedacore/keda/v2/pkg/metricsservice/api"
+)
+
+type metricMeta struct {
+	TriggerName      string
+	ScaledObjectName string
+	Namespace        string
+}
+
+type RawMetrics struct {
+	Meta   metricMeta
+	Values []external_metrics.ExternalMetricValue
+}
+
+type RawMetricSubscriptions struct {
+	id            string
+	subscriptions []metricMeta
+	rawMetrics    chan RawMetrics
+	done          chan bool
+}
+
+const (
+	// KEDA will keep this many measurements on the channel (each collected in 15s interval), then sendTimeout will start ticking
+	// if client isn't able to consume the message within that period, it will be automatically unsubscribed
+	rawMetricsChannelCapacity = 16
+
+	// timeout after which all the metrics for the subscriber will be unsubscribed (if these are not being consumed by client)
+	unsubscribeTimeout = 1 * time.Minute
+)
+
+func (h *scaleHandler) UnsubscribeMetric(_ context.Context, subscriber string, sor *api.ScaledObjectRef) bool {
+	toDel := toMetricMeta(sor)
+	log.Info("Unsubscribing metric", "subscriber", subscriber, "metric", toDel)
+	h.subsLock.Lock()
+	defer h.subsLock.Unlock()
+	if _, found := h.rawMetricsSubscriptions[subscriber]; !found || !h.isSubscribed(subscriber, sor) {
```

Comment:
UX nit: do we need to worry about "subscriber name" clash? e.g. if consumer runs in multiple replicas but has the same "subscriber name" configured? We could possibly leverage gRPC context `Peer` info and get the caller's IP address and use that as identification of the "subscriber".

https://pkg.go.dev/google.golang.org/grpc/peer#Peer

Reply:
nah, I'd keep the subscriberID as a logical thing and not derive it from IP or auth info, this is imho more universal